### ISSUE TITLE
textinput big list of todos

### ISSUE THREAD

**Issue by hanbollar** [Volumetrics-io/mrjs#590]
## necessary:
- [ ] https://github.com/Volumetrics-io/mrjs/pull/589
  - this is a blocker, once merged, then open up the mrtextarea and mrtextfield items on docs again

### both
- [ ] https://github.com/Volumetrics-io/mrjs/pull/588
- [ ] handle shared possible attributes:
  - [ ] `name`: The name associated with the <textarea/field> for form submission and backend processing.
  - [ ] `placeholder`: Provides a hint to the user about what they should type into the <textarea/field>.
  - [ ] `maxlength`: Specifies the maximum number of characters that the user can enter.
  - [ ] `readonly`: Makes the <textarea/field> uneditable, allowing the text to be only read, not modified.
  - [ ] `disabled`: Disables the text area so it cannot be interacted with or submitted.

### mr-textarea
- [ ] handle specific possible attributes:
  - [ ] `rows and cols`: These attributes control the size of the <textarea> in terms of the number of text rows and columns visible.
  - [ ] `wrap`: Controls how text is wrapped in the textarea, with values like soft and hard affecting form submission.
  - [ ] `overflowwrap` : Controls how wrap breaks, at whitespace characters or in the middle of words.
  - [ ] `whitespace` : Controls if text wraps with the overflowWrap feature or not.

### mr-textfield
- [ ] handle specific possible attributes
  - [ ] `value`: Sets the initial value of the input field.
  - [ ] `size`: Defines the width of the input field in characters.
  - [ ] `autofocus`: If present, specifies that the input field should automatically get focus when the page loads.
  - [ ] `required`: If present, specifies that the input field must be filled out before submitting the form. (the red error)
  - [ ] `autocomplete`: Enables or disables autocomplete suggestions for the input field. Values can be "on" or "off".
  - [ ] `pattern`: Specifies a regular expression that the input's value must match in order for the form to be submitted.
  - [ ] `title`: Provides a tooltip or advisory information about the input field when hovered over.
  - [ ] `id`: Specifies a unique ID for the input field, which can be used for targeting with CSS or JavaScript.

## nice to have:
_(todo: narrow down from these since theyre a bit more nebulous, what is for road to v1.0.0 and what is for future)_
- [ ] password field
- [ ] visual change if disabled
- [ ] visual change if readonly
- [ ] select cursor position (blocked by click events)
- [ ] Text selection
- [ ] copy/paste
- [ ] spell check
